### PR TITLE
No hardcoded fileids

### PR DESCRIPTION
**Pull request by Eldinnie**
Please do not merge, I want to see what travis thinks of this changes

### CODE REVIEW

**Comment by Eldinnie** [comment]
Ok, right now I expect the test_audio to throw an exception that should be fixed if the new audio-file is on master. Besides that I think this is about ready to merge

**Comment by Eldinnie** [comment]
Re: Travis:
Only the two tests I expected te fail do so `tests.test_audio.AudioTest` and `tests.test_audio.AudioTest` The rest is good.
Those error should be fixed when the new shorter audiofile is on the master branch

**Comment by Eldinnie** [comment]
Ok so I think it's about done now. `test_send_audio_mp3_url_file`, `test_send_audio_mp3_url_file_with_caption` and `test_send_sticker_from_url` are expected to fail until the files are adjusted/added on the master branch.

**Comment by Eldinnie** [comment]
On travis only expected tests failed.

File: tests/test_file.py [hunk 1]
```diff
@@ -135,14 +75,16 @@ def test_error_file_without_required_args(self):
...
+            self._bot.getFile(**json_dict)
+
 
     def test_equality(self):
-        a = telegram.File(self.audio_file_id, self._bot)
-        b = telegram.File(self.audio_file_id, self._bot)
-        c = telegram.File(self.audio_file_id, None)
-        d = telegram.File(self.document_file_id, self._bot)
-        e = telegram.Voice(self.audio_file_id, 0)
+        a = telegram.File("DOESNTMATTER", self._bot)
```
Comment (bomjacob) [inline tests/test_file.py]: Technically they do matter here. But it's fine :)

File: tests/test_photo.py [hunk 2]
```diff
@@ -18,111 +18,140 @@
...
+        self.assertNotEqual(photo.file_id, '')
+        self.assertEqual(photo.width, self.photo.width)
+        self.assertEqual(photo.height, self.photo.height)
+        self.assertEqual(photo.file_size, self.photo.file_size)
+
+    @flaky(3, 1)
+    @timeout(10)
+    def test_send_photo_url_png_file(self):
+        message = self._bot.sendPhoto(
+            photo='http://dummyimage.com/600x400/000/fff.png&text=telegram', chat_id=self._chat_id)
```
Comment (bomjacob) [inline tests/test_photo.py]: Could we maybe use a photo in our github here? Not a fan of depending on external sites (anymore than we already do)

File: tests/test_photo.py [hunk 3]
```diff
@@ -18,111 +18,140 @@
...
+
+        self.assertIsInstance(photo, telegram.PhotoSize)
+        self.assertIsInstance(photo.file_id, str)
+        self.assertNotEqual(photo.file_id, '')
+
+    @flaky(3, 1)
+    @timeout(10)
+    def test_send_photo_url_gif_file(self):
+        message = self._bot.sendPhoto(
+            photo='http://dummyimage.com/600x400/000/fff.png&text=telegram', chat_id=self._chat_id)
```
Comment (bomjacob) [inline tests/test_photo.py]: See other comment

File: tests/test_sticker.py [hunk 4]
```diff
@@ -18,90 +18,149 @@
...
-            'width': 90,
-            'height': 90,
-            'file_id': 'BQADAQADoQADHyP1B0mzJMVyzcB0Ag',
-            'file_size': 2364
-        }
-        self.emoji = telegram.Emoji.FLEXED_BICEPS
-        self.file_size = 39518
+    @classmethod
+    def setUpClass(cls):
+        cls.emoji = telegram.Emoji.FLEXED_BICEPS
```
Comment (bomjacob) [inline tests/test_sticker.py]: Please don't use the Emoji module as it's deprecated

File: tests/test_voice.py [hunk 5]
```diff
@@ -18,182 +18,153 @@
...
-        self.assertEqual(message.caption, self.caption)
-
-        voice = message.voice
-
-        self.assertTrue(isinstance(voice.file_id, str))
-        self.assertNotEqual(voice.file_id, '')
-        self.assertEqual(voice.duration, self.duration)
-        self.assertEqual(voice.mime_type, self.mime_type)
-        self.assertEqual(voice.file_size, self.file_size)
+        self.assertTrue(os.path.isfile('telegram.ogg'))
```
Comment (bomjacob) [inline tests/test_voice.py]: (applies to all downloads)
Should we check against the local file, that they're the same?

**Comment by tsnoam** [comment]
Merged according to discussion in the developers group.

**Pull request merged**